tensorflow_datasets/core/features/video_feature.py

```diff
@@ -197,26 +197,76 @@ def to_json_content(self) -> Json:
         'ffmpeg_extra_args': self._extra_ffmpeg_args
     }
 
+  def _generate_video_html(self, images, framerate=24, encoding_format='mp4') -> str:
+    """Converts sequence of images into video string."""
+    # CODE REDUNDANT: Could somehow use `_ffmpeg_decode`?
+    imgs = len(str(len(images)))+1  # Find number of digits in len to give names.
+    video_dir = tempfile.mkdtemp()
```

Comment:
To auto-cleanup the code, could you use `with tempfile. TemporaryDirectory():` contextmanager ?